### PR TITLE
moveaxis with sequence of dims and typing enhancements

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

### Description
The pytorch/numpy APIs supports a sequence of ints as `moveaxis` arguments,
this PR enhances the `moveaxis` in `monai/transforms/utils_pytorch_numpy_unification.py` so that it also accepts sequences.

The PR also enhances the typing annotations

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/utils.py [hunk 1]
```diff
@@ -411,7 +411,7 @@ def weighted_patch_samples(
 
 
 def correct_crop_centers(
-    centers: List[Union[int, torch.Tensor]],
+    centers: Union[NdarrayOrTensor, List[NdarrayOrTensor]],
```
Comment (wyli) [inline monai/transforms/utils.py:414]: Hi @Nic-Ma, I looked at the use of this `correct_crop_centers`, 

one is with a list input: https://github.com/Project-MONAI/MONAI/blob/523a04728a57719c5c598bd5708d211b28527cbc/monai/transforms/utils.py#L563-L564 the other is without:
https://github.com/Project-MONAI/MONAI/blob/523a04728a57719c5c598bd5708d211b28527cbc/monai/transforms/utils.py#L506-L508
any idea why there's the difference?

**Review by Nic-Ma — APPROVED** [review]
Looks good to me, thanks for the enhancement for `moveaxis`.

Thanks.

File: monai/transforms/utils.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/transforms/utils.py:414]: Hi @wyli ,

Thanks for your great finding!
I think the code was developed before we fixed the perf issue of GPU cached indices, etc.:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utils.py#L360
Actually, we don't need to convert to list of tensor and convert tensor to numpy. I submitted PR https://github.com/Project-MONAI/MONAI/pull/3652 to optimize the logic.
BTW, this PR looks good to me.

Thanks.

**Pull request merged**